### PR TITLE
test: Unskip ContentDialog leak test

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): #7860 Leak still persists on iOS. Can't investigate currently.

## PR Type

What kind of change does this PR introduce?

### <samp>🤖 Generated by Copilot at fcb9a63</samp>

Re-enabled a test case for memory leaks in `ContentDialog` on WebAssembly. The test case was previously disabled due to a bug.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml/Given_FrameworkElement_And_Leak.cs [hunk 1]
```diff
@@ -118,9 +118,7 @@ public class Given_FrameworkElement_And_Leak
 		[DataRow("UITests.Windows_UI_Xaml_Controls.ToolTip.ToolTip_LeakTest", 15)]
 		[DataRow("Uno.UI.RuntimeTests.Tests.Windows_UI_Xaml.Controls.Button_Command_Leak", 15)]
 		[DataRow("Uno.UI.RuntimeTests.Tests.Windows_UI_Xaml.Controls.ItemsControl_ItemsSource_Leak", 15)]
-#if !__WASM__ && !__IOS__ // Disabled - https://github.com/unoplatform/uno/issues/7860
 		[DataRow("Uno.UI.RuntimeTests.Tests.Windows_UI_Xaml.Controls.ContentDialog_Leak", 15)]
```
Comment (Youssef1313) [inline src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml/Given_FrameworkElement_And_Leak.cs:121]:
```suggestion
#if !__IOS__ // Disabled - https://github.com/unoplatform/uno/issues/7860
		[DataRow("Uno.UI.RuntimeTests.Tests.Windows_UI_Xaml.Controls.ContentDialog_Leak", 15)]
#endif
```

**Comment by jeromelaban** [comment]
@Youssef1313 let's retry one more to check a bit more on reproducibility.

**Comment by Youssef1313** [comment]
@jeromelaban Did the CI restart here? or do we need to give a restart to make sure the test is passing?

**Comment by jeromelaban** [comment]
It didn't, trying now

**Pull request closed without merging**